src/client/client.rs

```diff
@@ -972,6 +972,17 @@ impl Builder {
         self
     }
 
+    /// Set whether HTTP/1 connections will write header names as provided
+    /// at the socket level.
+    ///
+    /// Note that this setting does not affect HTTP/2.
+    ///
+    /// Default is false.
+    pub fn http1_preserve_header_case(&mut self, val: bool) -> &mut Self {
```

Comment:
Ah yes, so the original type in FFI I think just assumes, if you included the extension, even if empty, then you probably wanted the same returned in the `Response`. It's probably right to have this be an option too, just wanted to call out the difference in the review.

Reply:
Oh yea, I think the option is also *needed* for the server side, since you get the request before you could return a response with the extension set.